src/lib/tooltip/tooltip.ts

```diff
@@ -51,154 +75,217 @@ export class MdTooltip {
-    this._createOverlay();
+  /** Dispose the tooltip when destroyed */
+  ngOnDestroy() {
+    if (this._tooltipInstance) {
+      this._disposeTooltip();
+    }
   }
 
-  /**
-   * Remove overlay after tooltip's parent has been destroyed
-   */
-  ngOnDestroy() {
-    this.visible = false;
-    this._overlayRef.dispose();
-    this._overlayRef = null;
+  /** Shows the tooltip */
+  show(): void {
+    if (!this._tooltipInstance) {
+      this._createTooltip();
+    }
+
+    this._setTooltipMessage(this._message);
+    this._tooltipInstance.show(this._position);
   }
 
   /**
    * Create the overlay config and position strategy
+   * Hides the tooltip after the provided delay in ms. Defaults the delay to the material design
+   * prescribed delay time
    */
-  private _createOverlay() {
-    if (this._overlayRef) {
-      if (this.visible) {
-        // if visible, hide before destroying
-        this.hide();
-        this._createOverlay();
-      } else {
-        // if not visible, dispose and recreate
-        this._overlayRef.dispose();
-        this._overlayRef = null;
-        this._createOverlay();
-      }
-    } else {
-      let origin = this._getOrigin();
-      let position = this._getOverlayPosition();
-      let strategy = this._overlay.position().connectedTo(this._elementRef, origin, position);
-      let config = new OverlayState();
-      config.positionStrategy = strategy;
-
-      this._overlayRef = this._overlay.create(config);
+  hide(delay: number = TOOLTIP_HIDE_DELAY): void {
+    if (this._tooltipInstance) {
+      this._tooltipInstance.hide(delay);
     }
   }
 
-  /**
-   * Returns the origin position based on the user's position preference
-   */
+  /** Shows/hides the tooltip */
+  toggle(): void {
+    this._isTooltipVisible() ? this.hide() : this.show();
+  }
+
+  /** Returns true if the tooltip is currently visible to the user */
+  _isTooltipVisible(): boolean {
+    return this._tooltipInstance && this._tooltipInstance.isVisible();
+  }
+
+  /** Create the tooltip to display */
+  private _createTooltip(): void {
+    this._createOverlay();
+    let portal = new ComponentPortal(TooltipComponent, this._viewContainerRef);
+    this._tooltipInstance = this._overlayRef.attach(portal).instance;
+
+    // Dispose the overlay when finished the shown tooltip.
+    this._tooltipInstance.afterHidden().subscribe(() => {
+      this._disposeTooltip();
+    });
+  }
+
+  /** Create the overlay config and position strategy */
+  private _createOverlay(): void {
+    let origin = this._getOrigin();
+    let position = this._getOverlayPosition();
+    let strategy = this._overlay.position().connectedTo(this._elementRef, origin, position);
+    let config = new OverlayState();
+    config.positionStrategy = strategy;
+
+    this._overlayRef = this._overlay.create(config);
+  }
+
+  /** Disposes the current tooltip and the overlay it is attached to */
+  private _disposeTooltip(): void {
+    this._overlayRef.dispose();
```

Comment:
I am seeming some `TypeError: this._overlayRef is null` here, I think it originates from here, not 100% sure. Is this a known issue or is there need for one to be opened?